### DISCUSSION TITLE
How to run seed when deploy to heroku?

### DISCUSSION THREAD

**Discussion by toknT** [prisma/prisma#18057] [Q&A] [answered]
In my nest.js app there is a seed.ts generate the default administrator account, it has username and hashed password.

And this is the heroku procfile

```sh
web: npm run start:prod
release: npx prisma migrate deploy && npx prisma db seed
```

It return error when release
```sh
remote: Running seed command `ts-node prisma/seed.ts` ...
remote:
remote: An error occurred while running the seed command:remote: Error: Command failed with ENOENT: ts-node prisma/seed.tsremote: spawn ts-node ENOENTremote: Waiting for release.... failed.
```
Is there some way to let the seed.ts execed?(ps: current I am connect heroku-postgresql from local machine to manager online data, but I don't like it)

PS: It's a production enviroment  , so I don't like to install `typescript` and others  to `dependencies` not `devDependencies` in package.json.

**Comment by ludralph** [comment, marked as answer]
Hi @toknT 👋 

Seeding is primarily helpful when working in a development environment. If seed data is vital for the functioning of your application, it should be part of your application startup process, which should check for the existence of this data. Incorporating this seed into the application code would be a good idea.
An example of seeding data in application code can be found [here](https://github.com/prisma/prisma-examples/blob/428b7d79e1ad2e09d57ce607180a39e94217f564/deployment-platforms/heroku/src/index.js#L15)

**Reply by toknT** [reply]
Thanks, but I am using seeding for production enviroment not only development enviroment. Maybe create a [webhook](https://circleci.com/docs/webhooks/) to seed these data is the best choice in a project  coded by typescript and deployed on heroku.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
